src/navigations/HomeBottomTabs.js

```diff
@@ -16,188 +14,23 @@ import MemoNews from '../assets/icon/News';
-    permissions: {
-      alert: true,
-      badge: true,
-      sound: true
-    },
-    // Should the initial notification be popped automatically
-    // default: true
-    popInitialNotification: true,
-    requestPermissions: false
-  });
-
-  const pushNotifIos = (message) => {
-    if (__DEV__) {
-      console.log(message.messageId, 'message');
-    }
-    const {title, body} = message.notification;
-    PushNotificationIOS.addNotificationRequest({
-      title,
-      body,
-      id: message.messageId,
-      userInfo: message.data
-    });
-  };
-
-  const pushNotifAndroid = (remoteMessage) => {
-    const {title, body} = remoteMessage.notification;
-    PushNotification.localNotification({
-      id: '123',
-      title,
-      channelId: 'bettersosialid',
-      message: body,
-      data: remoteMessage.data
-    });
-  };
-
-  const createChannel = () => {
-    PushNotification.createChannel(
-      {
-        channelId: 'bettersosialid', // (required)
-        channelName: 'New Messages & Comments', // (required)
-        playSound: true, // (optional) default: true
-        soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function
-        importance: 4, // (optional) default: 4. Int value of the Android notification importance
-        vibrate: true // (optional) default: true. Creates the default vibration patten if true.
-      },
-      (created) => {
-        if (__DEV__) {
-          console.log(`createChannel returned '${created}'`);
-        }
-      } // (optional) callback returns whether the channel was created, false means it already existed.
-    );
-  };
-
-  const handlePushNotif = (remoteMessage) => {
-    const {data} = remoteMessage;
-    if (data.channel_type !== 3) {
-      if (isIos) {
-        pushNotifIos(remoteMessage);
-      } else {
-        pushNotifAndroid(remoteMessage);
-      }
-    }
-  };
-
-  const updateProfileAtomId = async () => {
-    const signedUserId = await getUserId();
-    const anonUserId = await getAnonymousUserId();
-    const token = TokenStorage.get(ITokenEnum.token);
-    const anonymousToken = TokenStorage.get(ITokenEnum.anonymousToken);
-
-    setProfileAtom({
-      anonProfileId: anonUserId,
-      signedProfileId: signedUserId,
-      token,
-      anonymousToken
-    });
-  };
-
-  React.useEffect(() => {
-    createChannel();
-    updateProfileAtomId();
-
-    messaging().setBackgroundMessageHandler(async (remoteMessage) => {
-      console.log('Message handled in the background!', remoteMessage);
-      handlePushNotif(remoteMessage);
-    });
-
-    const unsubscribe = messaging().onMessage((remoteMessage) => {
-      // eslint-disable-next-line no-unused-expressions
-      handlePushNotif(remoteMessage);
-    });
-    return () => {
-      unsubscribe();
-    };
-  }, []);
+  const navigation = useNavigation();
+  const {isLoadingFetchingChannelDetail} = usePushNotificationHook();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here involve removing a lot of code related to push notifications and replacing it with a hook `usePushNotificationHook`. This is a good move towards modularity and maintainability as it abstracts away the details of push notification handling into a separate hook. However, it's important to ensure that the new hook handles all the cases that were previously handled in this component.
```diff
- import profileAtom from '../atom/profileAtom';
- import TokenStorage, {ITokenEnum} from '../utils/storage/custom/tokenStorage';
- import {getAnonymousUserId, getUserId} from '../utils/users';
+ import usePushNotificationHook from '../hooks/core/push-notification/usePushNotificationHook';
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->